src/software/simulated_tests/kicker_double_touch.py

```diff
@@ -0,0 +1,95 @@
+        """Checks if the robot kicking the ball touches the ball again before another robot touches it
+
+        :param world: The world msg to validate
+        :returns: FAILING when the robot kicking the ball touches the ball again before another robot touches it
+                  PASSING when the robot kicking the ball does not touch the ball again before another robot touches it
+        """
+        ball_position = tbots_cpp.createPoint(world.ball.current_state.global_position)
+        if self.kick_position is None:
+            self.kick_position = ball_position
+        # quick passing check
+        if self.other_robot is not None:
+            return ValidationStatus.PASSING
+        # main check loop
+        for robot in world.friendly_team.team_robots:
+            if tbots_cpp.Robot(robot).isNearDribbler(ball_position, self.threshold):
+                if (
+                    self.kicker_robot == robot
+                    and self.other_robot is None
+                    and (ball_position - self.kick_position).length()
+                    > KICKOFF_DOUBLE_TOUCH_M
+                ):
+                    return ValidationStatus.FAILING
+                else:
+                    self.other_robot = robot
+                    return ValidationStatus.PASSING
+        return ValidationStatus.PASSING
+
+    def get_validation_geometry(self, world) -> ValidationGeometry:
+        """Returns the underlying geometry this validation is checking
+
+        :param world: The world msg to create validation geometry from
+        :returns: ValidationGeometry containing geometry to visualize
+
+        """
+        # assigns closest robot to ball as kicker if not already assigned
+        if self.kicker_robot is None:
+            ball_position = tbots_cpp.createPoint(world.ball.current_state.global_position)
+            dist = 1000
+            for robot in world.friendly_team.team_robots:
+                robot_position = tbots_cpp.Point(
+                    robot.current_state.global_position.x_meters,
+                    robot.current_state.global_position.y_meters,
+                )
+                if (ball_position - robot_position).length() < dist:
+                    dist = (ball_position - robot_position).length()
+                    self.kicker_robot = robot
+
+        return create_validation_geometry(
+            [
+                tbots_cpp.Circle(
+                    tbots_cpp.Point(
+                        self.kicker_robot.current_state.global_position.x_meters,
+                        self.kicker_robot.current_state.global_position.y_meters,
+                    ),
+                    0.5,
+                )
+            ]
+        )
+
+    def __repr__(self):
+        return "Checking that the robot kicking the ball is not touching the ball consecutively"
+
+
+(
+    KickerEventuallyNotDoubleTouch,
+    KickerEventuallyDoubleTouch,
+    KickerAlwaysNotDoubleTouch,
+    KickerAlwaysDoubleTouch,
```

Comment:
```suggestion
    _,
```
this validation isn't particularly useful to most people, so we can hide it